src/Controller/CommandLineController.ts

```diff
@@ -0,0 +1,47 @@
+import yargs from 'yargs'
+
+export interface JunknetArguments {
+	makefile: string
+	docker_image: string
+	target: string | null
+}
+
+/**
+ * Interpret a list of arguments, returning the relevant ones in the f
+ *
+ * @param argv - the invoking process's command-line arguments.
+ * @returns JunknetInterface object containing option values.
+ */
+export function interpretArgv(argv: readonly string[]): JunknetArguments {
+	// Use only the first two elements; Node.js appends extra elements to process.argv.
+	const yargsArgv = yargs(argv.slice(2)).options({
+		f: {
+			alias: 'makefile',
+			type: 'string',
+			default: 'Makefile',
+			desc: 'The Makefile to process',
+		},
+		i: {
+			alias: 'docker-image',
+			type: 'string',
+			default: 'ubuntu:18.04',
+			desc: 'The Docker Image to run',
+		},
+		t: {
+			alias: 'target',
+			type: 'string',
+			desc: 'The Makefile target to build',
+		},
```

Comment:
Shouldn't this be a positional argument, and allow dups? So you can do

    junknet-make all install clean

Instead of

    junknet-make -t all

Reply:
This makes sense -- we do want our interface to resemble GNU make. 

Now that I'm thinking more about interface compatibility, we should probably also: 

- Use short-option letters that aren't currently used by GNU make.
- Pass any extraneous options to `make`, so things like `make -B` won't cause our program to explode.  
- Search the current directory for `GNUmakefile`, `Makefile`, and `makefile`, using the first file it finds. 

I'll add these to the PR before merging.

Reply:
Take a look at the argument parser in [server.ts](https://github.com/cs130-w21/1/blob/master/bin/server.ts). I added a comment like this:

    // Modeled after sshd(8) where possible, see https://man.openbsd.org/sshd.

And made the options be a subset of sshd. That way in the future whoever wants to add something that's in sshd will use the same name and if it's not in sshd then they'll use a name that's not claimed.